src/xSTIR/cSTIR/include/sirf/STIR/stir_data_containers.h

```diff
@@ -438,6 +440,22 @@ namespace sirf {
 			ptr->fill(0.0f);
 			_data.reset(ptr);
 		}
+		PETAcquisitionDataInFile(std::unique_ptr<stir::ProjData> uptr_pd) : _owns_file(true)
```

Comment:
Isn't this a slightly weird constructor? If it's passed a `ProjDataInterfile` it will use it (and therefore if it's writable will modify), otherwise it will make a copy of the data with a temporary filename. Any modifications won't go to the argument it was constructed with.

I think this is surprising for the user.